sympy/solvers/solveset.py

```diff
@@ -1142,13 +1142,146 @@ def _is_exponential(f, symbol):
+def _solve_log(f, symbol):
+    r"""
+    Helper to solve logarithmic equations.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The logarithmic equation to be solved
+
+    symbol : Symbol
+        The variable in which the equation is solved
+
+    Returns
+    =======
+
+    An improved equation containg a single instance of log.
+
+    `None`:
+        If the equation does not changes.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, S, log
+    >>> from sympy.solvers.solveset import _solve_log as solve_log
+    >>> x = symbols('x')
+    >>> f = log(x - 3) + log(x + 3)
+    >>> solve_log(f, x)
+    log((x - 3)*(x + 3))
+
+    * Proof of correctness
+
+    A logarithm is another way to write exponent and is defined by
+
+    .. math:: {log_b x} = y \enspace if \enspace b^y = x
+
+    When one side of the equation contains a single logarithm, the
+    equation can be solved by rewriting the equation as an equivalent
+    exponential equation as defined above. But if one side contains
+    more than one logarithm, we need to use the properties of logarithm
+    to condense it into a single logarithm.
+
+    Take for example
+
+    .. math:: log(2*x) - 15 = 0
+
+    contains single logarithm, therefore we can directly rewrite it to
+    exponential form as
+
+    .. math:: x = exp(15)/2
+
+    But if the equation has more than one logarithm as
+
+    .. math:: log(x - 3) + log(x + 3) = 0
+
+    we use log identities to convert into a reduced form
+
+    Using,
+
+    .. math:: log(a) + log(b) = log(a*b)
+
+    therefore the equation becomes,
+
+    .. math:: log((x - 3)*(x  + 3))
+
+    This equation contains one logarithm and can be solved by rewriting
+    to exponents.
+    """
+
+    lhs, rhs_s = invert_real(f, 0, symbol)
+    rhs = rhs_s.args[0]
+    new_f = None
+
+    new_lhs = logcombine(lhs, force=True)
+    if new_lhs != lhs:
+        new_f = new_lhs - rhs
+
+    return new_f
+
+
+def _is_logarithmic(f, symbol):
+    r"""
+    Helper to check if the given equation is logarithmic or not.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The equation to be checked
+
+    symbol : Symbol
+        The variable in which the equation is checked
+
+    Returns
+    =======
+
+    `True` if the equation is logarithmic otherwise `False`.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, tan, log
+    >>> from sympy.solvers.solveset import _is_logarithmic as check
+    >>> x = symbols('x')
+    >>> check(log(x + 2) - log(x + 3), x)
+    True
+    >>> check(tan(log(2*x)), x)
+    False
+
+    * Philosophy behind the helper
+
+    The function extracts each term of the equation and checks if it
+    it is an instance of `log` w.r.t `symbol`
+    """
+
+    def check_log(arg, symbol):
+        if isinstance(arg, log) and (
+                symbol in arg.free_symbols):
+            return True
+
+    expr_args = _term_factors(f)
+    for expr_arg in expr_args:
```

Comment:
suggest combining into `for expr_arg in _term_factors(f):`

Reply:
Just used to improve readability. One could at once get what is returned from `_term_factors`. Should I change it?